### ISSUE TITLE
Ground truth data prep

### ISSUE THREAD

**Issue by marcinkusz** [zhaihongjia/PanoGS#5]
Hi, cool work! I was wondering if you could comment on how you prepped the GT data more. 
I am currently trying to reproduce your work on my own data and am wondering exactly how you processed the semantic label encodings in `datasets/scannet/SCANNET_LABELS_20.npy`

I realize you used an OpenClip model, but I tried using some ViT-B/16 model versions from open_clip, openai/clip and lseg to encode these scannet labels:
```python
text_labels = ['wall', 'floor', 'chair', 'table', 'desk', 'bed', 'bookshelf', 
               'sofa', 'sink', 'bathtub', 'toilet', 'curtain', 'counter', 'door', 'window', 
               'shower curtain', 'refridgerator', 'picture', 'cabinet', 'otherfurniture']
``` 
and cannot seem to reproduce the exact results in the file `SCANNET_LABELS_20.npy`
Could you shortly expand on how you did it? Did you run any pre- or postprocessing steps? Did you template the labels or just use raw single word prompts? Which model and library did you use exactly?

**Comment by zhaihongjia** [comment]
Hello, I use the method in OpenScene to generate the TEXT feature. You can try the following codes:

```
import os
import numpy as np
import clip

def extract_clip_feature(labelset, model_name="ViT-B/32"):
    print("Loading CLIP {} model...".format(model_name))
    clip_pretrained, _ = clip.load(model_name, device='cuda', jit=False)

    if isinstance(labelset, str):
        lines = labelset.split(',')
    elif isinstance(labelset, list):
        lines = labelset
    else:
        raise NotImplementedError

    labels = []
    for line in lines:
        label = line
        labels.append(label)
    text = clip.tokenize(labels)
    text = text.cuda()
    text_features = clip_pretrained.encode_text(text)
    text_features = text_features / text_features.norm(dim=-1, keepdim=True)

    return text_features

def encode_text_feature(labelset, setype='scannet'):
    if setype == 'scannet':
        labelset = [ "a " + label + " in a scene" for label in labelset]
        labelset[-1] = 'other'

    elif setype == 'replica':
        labelset = [ "a " + label + " in a scene" for label in labelset]
        labelset[-2] = 'other'
    else:
        pass
    
    text_features = extract_clip_feature(labelset)
    return text_features


SCANNET_LABELS_20 = ['wall', 'floor', 'cabinet', 'bed', 'chair', 'sofa',
                     'table', 'door', 'window', 'bookshelf', 'picture','counter', 'desk', 'curtain', 'refrigerator', 'shower curtain',
                     'toilet', 'sink', 'bathtub', 'otherfurniture']

text_features = encode_text_feature(SCANNET_LABELS_20, 'scannet')
print(text_features.shape)

np.save(os.path.join(query_path, 'text_feat.npy'), text_features.detach().cpu().numpy())
```

If you have any other questions, please give me feedback.

**Comment by marcinkusz** [comment]
Thank you for the quick answer, you comment was the last step I needed to get it to work!
